pkg/storage/utils/decomposedfs/spaces.go

```diff
@@ -427,6 +427,36 @@ func (fs *Decomposedfs) ListStorageSpaces(ctx context.Context, filter []*provide
 
 }
 
+func (fs *Decomposedfs) checkNodePermissions(ctx context.Context, requestedUserID string, unrestricted bool) bool {
+	authenticatedUserID := ctxpkg.ContextMustGetUser(ctx).GetId().GetOpaqueId()
+	canListAllSpaces := fs.canListAllSpaces(ctx)
+	switch {
+	case canListAllSpaces && requestedUserID == spaceTypeAny:
+		return true
+	case canListAllSpaces && authenticatedUserID != requestedUserID:
+		// as admin you have to be able to see other users spaces
+		return false
+	case canListAllSpaces, unrestricted:
+		// standard case, requesting user needs to check space permissions
+		return false
+	}
+	return true
+}
+
+func (fs *Decomposedfs) canListSpacesOfRequestedUser(ctx context.Context, requestedUserID string) bool {
+	authenticatedUserID := ctxpkg.ContextMustGetUser(ctx).GetId().GetOpaqueId()
+	switch {
+	case requestedUserID == spaceTypeAny:
```

Comment:
Makes no sense imho. A userID and a spaceType can't match. Maybe rename `spaceTypeAny`?